code/game/turfs/simulated/floor/lava.dm

```diff
@@ -121,8 +121,20 @@
 			new /obj/structure/lattice/lava(locate(x, y, z))
 		else
 			to_chat(user, "<span class='warning'>You need one rod to build a heatproof lattice.</span>")
-			return
-
+		return
+	if(istype(C, /obj/item/stack/tile/plasteel))
+		var/obj/structure/lattice/L = locate(/obj/structure/lattice/lava, src)
+		if(L)
```

Comment:
Last thing, could you make this an early return?

Reply:
sure